### DISCUSSION TITLE
Dev server not starting with react plugin

### DISCUSSION THREAD

**Discussion by iamravisingh** [web-infra-dev/rsbuild#2837] [Q&A] [answered]
<img width="1728" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/34270887/62ad7ca5-5aec-4aa1-88cd-1248224f7524">

Hello Everyone, 

I am trying to migrate from CRA to rspack. I followed the amazing docs to use rsbuild for my use case. However, currently stuck on some local error for react plugin attached as a screenshot. Initially it did worked, later with some changes it started failing with attached error. I have also added the version of rsbuild which I am currently using.

 Additionally, I have also observed, plugin-react is yelling with ts error `Rspack.SwcLoaderTransformConfig['react']` is failing with below error. Does this somewhere connected?
 
<img width="1399" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/34270887/29e0d84d-6272-47c9-868a-3135f7a07956">

I might be missing something small, any help is appreciated. Please find the below config file and package details.

 ```
 npmPackages:
    @rsbuild/core: ^1.0.0-alpha.5
    @rsbuild/plugin-eslint: ^1.0.0
    @rsbuild/plugin-react: ^1.0.0-alpha.5
    @rsbuild/plugin-sass: ^1.0.0-alpha.5

```
**rsbuild.config.ts**
```
/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig } from "@rsbuild/core";
import { pluginReact } from "@rsbuild/plugin-react";
import { pluginEslint } from "@rsbuild/plugin-eslint";
import { pluginSass } from '@rsbuild/plugin-sass';


export default defineConfig({
    plugins: [
        pluginReact({
            enableProfiler: true,
        }),
        pluginEslint(),
        pluginSass()
    ],
    dev: {
        client:{
            port: "3000"
        },
        progressBar: {
            id: "Build progress"
        }
    }
});
```

**Comment by 9aoy** [comment, marked as answer]
Hey, according to the error message you provided,  I suspect that you have installed an unexpected version of rsbuild (rsbuild 1.0.0 is a deprecated version 😭 ), maybe you should lock rsbuild to `1.0.0-alpha.5`.

```diff
npmPackages:
-   @rsbuild/core: ^1.0.0-alpha.5
-   @rsbuild/plugin-eslint: ^1.0.0
-   @rsbuild/plugin-react: ^1.0.0-alpha.5
-   @rsbuild/plugin-sass: ^1.0.0-alpha.5
+   @rsbuild/core: 1.0.0-alpha.5
+   @rsbuild/plugin-eslint: 1.0.0-alpha.5
+   @rsbuild/plugin-react: 1.0.0-alpha.5
+   @rsbuild/plugin-sass: 1.0.0-alpha.5
```

**Reply by iamravisingh** [reply]
Ah! Damn, I knew I was missing something small. @9aoy thanks for the quick response. 
I have locked the version for the rsbuild packages and things are working fine.
